adminSiteClient/CreateDataInsightModal.tsx

```diff
@@ -0,0 +1,935 @@
+import {
+    faCheck,
+    faSpinner,
+    faRobot,
+    faCheckCircle,
+    faCircleXmark,
+    faDownload,
+} from "@fortawesome/free-solid-svg-icons"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import {
+    Form,
+    Modal,
+    Input,
+    Button,
+    Space,
+    Flex,
+    Select,
+    FormItemProps,
+} from "antd"
+import { ValidateStatus } from "antd/es/form/FormItem"
+import { useCallback, useContext, useEffect, useMemo, useState } from "react"
+import cx from "classnames"
+import {
+    fetchFigmaProvidedImageUrl,
+    ImageUploadResponse,
+    uploadImageFromSourceUrl,
+} from "./imagesHelpers"
+import { AdminAppContext } from "./AdminAppContext"
+import { GRAPHER_DYNAMIC_THUMBNAIL_URL } from "../settings/clientSettings"
+import { LoadingImage } from "./ReuploadImageForDataInsightModal"
+import { ApiChartViewOverview } from "../adminShared/AdminTypes"
+import { downloadImage, isEmpty, RequiredBy } from "@ourworldindata/utils"
+import { match } from "ts-pattern"
+
+const DEFAULT_RUNNING_MESSAGE: Record<Task, string> = {
+    createDI: "Creating data insight...",
+    uploadImage: "Uploading image...",
+    loadFigmaImage: "Loading Figma image...",
+    suggestAltText: "Suggesting alt text...",
+} as const
+
+const DEFAULT_SUCCESS_MESSAGE: Record<Task, string> = {
+    createDI: "Data insight created successfully",
+    uploadImage: "Image uploaded successfully",
+    loadFigmaImage: "Figma image loaded successfully",
+    suggestAltText: "Alt text suggested successfully",
+} as const
+
+const DEFAULT_ERROR_MESSAGE: Record<Task, string> = {
+    createDI: "Data insight creation failed",
+    uploadImage: "Uploading image failed",
+    loadFigmaImage: "Loading Figma image failed",
+    suggestAltText: "Suggesting alt text failed",
+} as const
+
+type Task = "createDI" | "uploadImage" | "loadFigmaImage" | "suggestAltText"
+
+type Progress =
+    | { status: "idle" }
+    | { status: "running"; message: string }
+    | { status: "complete"; success: boolean; message: string }
+
+type FormFieldName =
+    | "title"
+    | "authors"
+    | "grapherUrl"
+    | "narrativeChart"
+    | "figmaUrl"
+    | "imageFilename"
+    | "imageAltText"
+type ImageFormFieldName = "imageFilename" | "imageAltText"
+
+type FormData = Partial<Record<FormFieldName, string>>
+type FormDataWithTitle = RequiredBy<FormData, "title">
+type FormDataWithImageFilename = RequiredBy<FormData, "imageFilename">
+
+type DataInsightCreationResponse =
+    | { success: true; gdocId: string }
+    | { success: false; errorMessage: string }
```

Comment:
I think this okay because you're actually handling them here, but possibly for consistency's sake, we should just throw them as JsonErrors (See https://github.com/owid/owid-grapher/pull/4564)

But maybe `admin.requestJSON` will intercept that and put up the ugly error modal, in which case, the way you're doing it is definitely superior 🙂